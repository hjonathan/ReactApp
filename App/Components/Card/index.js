import React, {Component} from 'react';
import {View} from 'react-native';
import Styles from './styles';

/**
 * Class Card
 * Example:
 * import Card from '../Card';
 * render() {
 *      return(
 *          <Card>
 *              {content}
 *          </Card>
 *      ),
 * }
 */
class CardView extends Component {
    /**
     * Constructor cardView
     * @param props
     */
    constructor(props) {
        super(props);
    };

    /**
     * Render
     * @param data
     * @returns {*}
     */
    render() {
        return (
            <View style={Styles.mainContainer} elevation={3}>
                {this.props.children}
            </View>
        );
    }
}

export default CardView;