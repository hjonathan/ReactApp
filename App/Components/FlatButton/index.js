import React, {Component} from "react";
import {
    Text,
    TouchableOpacity,
    View
} from "react-native";
import Styles from "./styles";

/**
 * Class FlatButton
 * Example:
 *
 * import FlatButton from '../FlatButton';
 * render() {
 *      return (
 *          <FlatButton
 *              params={{radius: 4, border: 1, color: '#fff'}}
 *              onPress={() => {}}
 *              title="Title test"
 *          />
 *      );
 * }
 */
export default class Button extends Component {
    /**
     * Constructor FlatButton
     * @param props
     */
    constructor (props) {
        super(props);
        this.state = {
            title: (props && props.title) || "test"
        };
    }

    /**
     * Returns custom button
     * @param params
     * @returns {*}
     */
    renderCustomButton (params) {
        return (
            <View style={{
                borderRadius: params.radius,
                borderWidth: params.border,
                borderColor: params.color,
                alignItems: "center",
                paddingHorizontal: 50,
                paddingVertical: 10
            }}
            >
                <Text style={{color: params.color, fontWeight: "bold"}}>
                    {this.state.title}
                </Text>
            </View>
        );
    }

    /**
     * Returns default button
     * @returns {*}
     */
    renderDefaultButton () {
        return (
            <View style={Styles.button}>
                <Text style={Styles.text}>
                    {this.state.title}
                </Text>
            </View>
        );
    }

    /**
     * Renders Component
     * @param onPress
     * @returns {*}
     */
    render ({onPress} = this.props) {
        return (
            <TouchableOpacity onPress={onPress}>
                {
                    (Object.prototype.hasOwnProperty.call(this.props, "params") && this.props.params)
                        ? this.renderCustomButton(this.props.params) : this.renderDefaultButton()
                }
            </TouchableOpacity>
        );
    }
}
