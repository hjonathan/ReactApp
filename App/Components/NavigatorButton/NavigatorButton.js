import React, {Component} from 'react';
import {
    Text,
    TouchableOpacity,
    View
} from 'react-native';
import Styles from './styles';
import PropTypes from 'prop-types';
import * as Res from '../../Assets/resources';
import Icon from 'react-native-vector-icons/FontAwesome';

export default class NavigatorButton extends Component {
    /**
     * Costructor
     * @param {*} props
     */
    constructor(props) {
        super(props);
    }

    /**
     * Creates the component that must be setted into the modal element.
     */
    _renderNavigatorButton = () => {
        const {
            title,
            icon,
            borderWidth
        } = this.props;
        return (
            <TouchableOpacity onPress={this.props.onPress}>
                <View style={Styles.navTitleBox}>
                    <View style={Styles.navTitle}>
                        <Icon name={icon} style={{color: Res.colors.white, fontSize: 24}}/>
                    </View>
                    <View style={Styles.navOption}>
                        <Text style={Styles.navTitleText}>{title}</Text>
                    </View>
                </View>
            </TouchableOpacity>
        );
    };

    /**
     * Render and create the modal element
     */
    render() {
        return (
            this._renderNavigatorButton()
        );
    }
}
/**
 * Initialize the prop to be mapped and validated into the component
 *
 * @ignore
 */
NavigatorButton.propTypes = {
    title: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired,
    onPress: PropTypes.func.isRequired,
    borderWidth: PropTypes.number,
}
/**
 * Sets the default values
 */
NavigatorButton.defaultProps = {
    title: '',
    icon: '',
    borderWidth: 1
}