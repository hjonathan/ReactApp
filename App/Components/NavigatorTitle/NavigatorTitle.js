import React, {Component} from "react";
import {
    Text,
    View
} from "react-native";
import PropTypes from "prop-types";
import Styles from "./styles";

/**
 * Class FlatButton
 * usage Example:
 *
 * import NavigatorTitle from '../Prompt';
 *   <NavigatorTitle
 *       title='Inbox'
 *       Number='70'/>
 *
 */
export default class NavigatorTitle extends Component {
    /**
     * Creates the component that must be setted into the modal element.
     * considere that will be used to enter into a modal element.
     * @private
     */
    _renderNavigatorTitle = () => {
        const {
            title,
            number,
            borderWidth
        } = this.props;
        return (
            <View style={Styles.navTitleBox}>
                <View style={Styles.navTitle}>
                    <Text style={Styles.navTitleText}>{title}</Text>
                </View>
                <View style={Styles.navCounter}>
                    <Text style={[Styles.navCounterText, {borderWidth}]}>{number}</Text>
                </View>
            </View>
        );
    };

    /**
     * Render and create the modal element, an fill with the dialog view previowsly created
     * in  _renderNavigatorTitle method
     */
    render () {
        return (
            this._renderNavigatorTitle()
        );
    }
}
/**
 * Initialize the prop to be mapped and validated into the component
 *
 * @ignore
 */
NavigatorTitle.propTypes = {
    title: PropTypes.string.isRequired,
    number: PropTypes.number.isRequired,
    borderWidth: PropTypes.number
};
/**
 * Sets the default values
 */
NavigatorTitle.defaultProps = {
    borderWidth: 1
};
