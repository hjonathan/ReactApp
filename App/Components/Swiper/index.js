import React, {Component} from 'react';
import {
    Dimensions,
    ScrollView,
    View,
    Image
} from 'react-native';
import Styles from './styles';
import * as Res from '../../Assets/resources';

const {width} = Dimensions.get('window');

/**
 * Class Swiper
 * Example:
 * import Swiper from '../Swiper';
 * ...
 * render() {
 *      return(
 *          <Swiper>
 *              <View>
 *                  ...content
 *              </View>
 *              <View>
 *                  ...content
 *              </View>
 *              <View>
 *                  ...content
 *              </View>
 *          </Swiper>
 *      );
 * }
 */
class Swiper extends Component {
    /**
     * Constructor Swiper
     * @param props
     */
    constructor(props) {
        super(props);
        this.state = {
            currentIndex: 0,
            totalSlide: (this.props && this.props.children && this.props.children.length) || 0,
            widthScreen: width
        };
        this.onChangeLayout = this.onChangeLayout.bind(this);
    }

    /**
     * Updates current position
     * @param event
     */
    updateIndex(event) {
        let x = event.nativeEvent.contentOffset.x,
            currentIndex = parseInt(parseInt(x) / parseInt(this.itself.state.widthScreen));
        this.itself.setState({currentIndex});
        this.itself.props.onChangeIndex(currentIndex);
    }

    /**
     * Fires when the scroll begin
     * @param {*} event 
     */
    onMomentumScrollBegin(event) {
        this.itself.props.onMomentumScrollBegin(event);
    }

    /**
     * Gets current screen's width and updates withScreen state
     * @param e
     */
    onChangeLayout(e) {
        const {width} = Dimensions.get('window');
        this.setState({widthScreen: width});
    };

    /**
     * Render slides
     * @returns {*}
     */
    renderSlide() {
        if (this.state.totalSlide) {
            return (
                <ScrollView
                    ref="_scrollView"
                    pagingEnabled={true}
                    horizontal={true}
                    showsHorizontalScrollIndicator={false}
                    showsVerticalScrollIndicator={false}
                    onMomentumScrollEnd={this.updateIndex}
                    itself={this}
                    onLayout={this.onChangeLayout}
                    onMomentumScrollBegin={this.onMomentumScrollBegin}
                >
                    {this.props.children.map((child, index) =>
                        <View key={index}>
                            {child}
                        </View>
                    )}
                </ScrollView>
            );
        }
    };

    /**
     * Render dots
     */
    renderPagination() {
        let dots = [],
            key;
        if (this.state.totalSlide >= 2) {
            let activeDot = <View style={[Styles.dot, Styles.activeDot]}/>,
                dot = <View style={Styles.dot}/>;
            for (key = 0; key < this.state.totalSlide; key += 1) {
                dots.push(key === this.state.currentIndex ?
                    React.cloneElement(activeDot, {key}) :
                    React.cloneElement(dot, {key})
                );
            }
            return (
                <View pointerEvents="none" style={[Styles.pagination, {width: this.state.widthScreen}]}>
                    {dots}
                </View>
            );
        }
    }

    /**
     * Render component
     * @returns {*}
     */
    render() {
        return (
            <View style={[Styles.mainContainer]}>
                <View style={Styles.logoContainer}>
                    <Image
                        source={Res.images.logo_white}
                        style={Styles.logo}/>
                </View>
                {this.renderSlide()}
                {this.renderPagination()}
            </View>
        );
    }
}

export default Swiper;