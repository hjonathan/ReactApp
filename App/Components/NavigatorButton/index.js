import NavigatorButton from './NavigatorButton';
export default NavigatorButton;