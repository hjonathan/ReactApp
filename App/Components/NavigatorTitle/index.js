import NavigatorTitle from './NavigatorTitle';
export default NavigatorTitle;
