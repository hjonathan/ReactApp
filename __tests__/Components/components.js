import "react-native";
import React from "react";
import FlatButton from "../../App/Components/FlatButton";
import Swiper from "../../App/Components/Swiper";
import Card from "../../App/Components/Card";
import ItemList from "../../App/Components/ItemList";
import NavigatorTitle from "../../App/Components/NavigatorTitle";
import NavigatorButton from "../../App/Components/NavigatorButton";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";

import renderer from "react-test-renderer";
import {CASE} from "../../App/Libs/Const";

describe("Component >>> ItemList", () => {
    const parent = {
        navigation: {
            dispatch: () => {},
            state: {
                routeName: CASE.PARTICIPATED
            }
        }
    };
    const data = {
        caseId: "",
        caseNumber: 0,
        caseTitle: "",
        dueDate: "1995-12-25",
        process: {
            name: ""
        },
        task: {
            name: ""
        },
        prevUser: {
            fullName: "",
            firstName: "",
            lastName: ""
        },
        userPhoto: ""
    };
    const dataDraft = {
        caseId: "",
        caseNumber: 0,
        caseTitle: "",
        dueDate: "1995-12-25",
        process: {
            name: ""
        },
        task: {
            name: ""
        },
        currentUser: {
            fullName: "",
            firstName: "",
            lastName: ""
        },
        userPhoto: ""
    };
    const dataIncomplete = {
        caseId: "",
        caseNumber: 0,
        caseTitle: "",
        dueDate: "1995-12-25",
        process: {
            name: ""
        },
        task: {
            name: ""
        },
        userPhoto: ""
    };
    let itemList,
        itemListDraft,
        itemListIncomplete;

    beforeEach(() => {
        itemList = shallow(<ItemList parent={parent} data={data} />);
        itemListDraft = shallow(<ItemList parent={parent} data={dataDraft} />);
        itemListIncomplete = shallow(<ItemList parent={parent} data={dataIncomplete} />);
    });

    it("Render ItemList", () => {
        expect(toJson(itemList)).toMatchSnapshot();
        expect(toJson(itemListDraft)).toMatchSnapshot();
    });

    it("ItemList test onPress", () => {
        expect(toJson(itemList.props("children").onPress())).toMatchSnapshot();
    });

    it("ItemList test _onLoadImage", () => {
        expect(toJson(itemList.instance()._onLoadImage())).toMatchSnapshot();
    });

    it("ItemList test _getCapitalLetter", () => {
        expect(toJson(itemList.instance()._getCapitalLetter())).toMatchSnapshot();
        expect(toJson(itemListDraft.instance()._getCapitalLetter())).toMatchSnapshot();
        expect(toJson(itemListIncomplete.instance()._getCapitalLetter())).toMatchSnapshot();
    });

    it("ItemList test onPress controls", () => {
        let cardView = itemList.props("children").children,
            cardContent = cardView.props.children,
            secondSection = cardContent.props.children[1],
            controlGroup = secondSection.props.children[1],
            iconButtons = controlGroup.props.children[1],
            btnHistory = iconButtons.props.children[0].props,
            btnNotes = iconButtons.props.children[1].props;
        expect(toJson(btnHistory.onPress())).toMatchSnapshot();
        expect(toJson(btnNotes.onPress())).toMatchSnapshot();
    });
});

it("Render FlatButton", () => {
    const button = shallow(<FlatButton />);
    expect(toJson(button)).toMatchSnapshot();
});

it("Render Swiper", () => {
    const swiper = renderer.create(<Swiper />).toJSON();
    expect(swiper).toMatchSnapshot();
});

it("Render Card", () => {
    const card = renderer.create(<Card />).toJSON();
    expect(card).toMatchSnapshot();
});

it("Validator On Time", () => {
    const valuesTest = {
        dueDateFormated: new Date("2018-01-17T13:30:13+00:00"),
        sentDateFormated: new Date("2018-01-17T11:54:13+00:00"),
        now: new Date("2018-01-16T13:30:13+00:00")
    };
    const wrapper = shallow(<ItemList />);
    expect(wrapper.instance().getStatus(valuesTest)).toEqual("ontime");
});
it("Validator dates Overdue", () => {
    const valuesTest = {
        dueDateFormated: new Date("2018-01-15T13:30:13+00:00"),
        sentDateFormated: new Date("2018-01-17T11:54:13+00:00"),
        now: new Date("2018-01-16T13:30:13+00:00")
    };
    const wrapper = shallow(<ItemList />);
    expect(wrapper.instance().getStatus(valuesTest)).toEqual("overdue");
});

it("Validator At Risk", () => {
    const valuesTest = {
        dueDateFormated: new Date("2018-01-17T16:30:13+00:00"),
        sentDateFormated: new Date("2018-01-16T11:54:13+00:00"),
        now: new Date("2018-01-16T13:30:13+00:00")
    };
    const wrapper = shallow(<ItemList />);
    expect(wrapper.instance().getStatus(valuesTest)).toEqual("atrisk");
});

it("Property maps - On time", () => {
    const wrapper = shallow(<ItemList />);
    expect(wrapper.instance().statusSettingMap.ontime).toEqual({
        color: "#1FBC99",
        textStatus: "lists_activity_due_date_on_time"
    });
});

it("Property maps - Overdue", () => {
    const wrapper = shallow(<ItemList />);
    expect(wrapper.instance().statusSettingMap.overdue).toEqual({
        color: "#E14333",
        textStatus: "lists_activity_due_date_overdue"
    });
});

it('Property maps - At risk', () => {
    const wrapper = shallow(<ItemList/>);
    expect(wrapper.instance().statusSettingMap['atrisk']).toEqual({
        "color": "#F1C500",
        "textStatus": "lists_activity_due_date_at_risk"
    });
});

it("Render NavigatorTitle", () => {
    const navigatorTitle = renderer
        .create(<NavigatorTitle title="test" number={5} />)
        .toJSON();
    expect(navigatorTitle).toMatchSnapshot();
});

it("Render NavigatorButton", () => {
    const navigatorButton = shallow(<NavigatorButton title="test" icon="sign-out" onPress={() => {}} />);
    expect(toJson(navigatorButton)).toMatchSnapshot();
});
