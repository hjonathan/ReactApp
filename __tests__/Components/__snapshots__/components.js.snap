// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component >>> ItemList ItemList test _getCapitalLetter 1`] = `undefined`;

exports[`Component >>> ItemList ItemList test _getCapitalLetter 2`] = `undefined`;

exports[`Component >>> ItemList ItemList test _getCapitalLetter 3`] = `undefined`;

exports[`Component >>> ItemList ItemList test _onLoadImage 1`] = `undefined`;

exports[`Component >>> ItemList ItemList test onPress 1`] = `undefined`;

exports[`Component >>> ItemList ItemList test onPress controls 1`] = `undefined`;

exports[`Component >>> ItemList ItemList test onPress controls 2`] = `undefined`;

exports[`Component >>> ItemList Render ItemList 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <CardView>
    <Component
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 6,
          "overflow": "hidden",
        }
      }
    >
      <Component
        style={
          Object {
            "flexDirection": "row",
            "paddingBottom": 8,
            "paddingHorizontal": 16,
            "paddingTop": 16,
          }
        }
      >
        <Component
          style={
            Object {
              "width": "80%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "#000",
                "fontSize": 18,
                "fontWeight": "bold",
                "padding": 1,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "#000",
                "fontSize": 16,
              }
            }
          >
             - 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "gray",
                "fontSize": 12,
              }
            }
          >
            sent_on
            1995-12-25 00:00:00
          </Text>
        </Component>
        <Component
          style={
            Object {
              "alignItems": "flex-end",
              "width": "20%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#2796ea",
                "fontSize": 10,
                "fontWeight": "bold",
                "paddingHorizontal": 3,
                "textAlign": "right",
              }
            }
          >
            0
          </Text>
        </Component>
      </Component>
      <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#f5f5f5",
            "flexDirection": "column",
            "paddingHorizontal": 10,
          }
        }
      >
        <Component
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingTop": 5,
              "width": "100%",
            }
          }
        >
          <Component
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Component
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#1F6CBC",
                    "borderRadius": 100,
                    "height": 36,
                    "justifyContent": "center",
                    "width": 36,
                  },
                ]
              }
            >
              <Image
                onLoad={[Function]}
                source={
                  Object {
                    "uri": "data:image/gif;base64,",
                  }
                }
                style={
                  Array [
                    Object {
                      "borderRadius": 100,
                      "height": 36,
                      "resizeMode": "cover",
                      "width": 36,
                    },
                    Object {
                      "display": "none",
                    },
                  ]
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "display": "flex",
                  }
                }
              />
            </Component>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000",
                  "paddingHorizontal": 8,
                  "paddingVertical": 6,
                }
              }
            >
               
            </Text>
          </Component>
          <Component
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <Component
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 3,
                    "flexDirection": "row",
                    "margin": 1,
                    "padding": 3,
                  },
                  Object {
                    "backgroundColor": "#E14333",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  }
                }
              >
                lists_activity_due_date_overdue
              </Text>
            </Component>
          </Component>
        </Component>
        <Component
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "paddingTop": 5,
                "width": "100%",
              },
              Object {
                "paddingTop": 2,
              },
            ]
          }
        >
          <Component
            style={
              Object {
                "alignItems": "center",
                "display": "none",
              }
            }
          >
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingTop": 5,
                }
              }
            >
              <Icon
                allowFontScaling={false}
                name="check-circle"
                size={12}
                style={
                  Object {
                    "color": "#1FBC99",
                    "fontSize": 24,
                  }
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#1FBC99",
                      "justifyContent": "flex-start",
                      "paddingHorizontal": 8,
                      "paddingVertical": 6,
                    },
                    Object {
                      "color": "#1FBC99",
                    },
                  ]
                }
              >
                saved_on_Draft_new_activity_string
              </Text>
            </Component>
          </Component>
          <Component
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Icon
                allowFontScaling={false}
                color="#000"
                name="history"
                size={30}
                style={
                  Object {
                    "paddingLeft": 4,
                  }
                }
              />
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Icon
                allowFontScaling={false}
                color="#000"
                name="sticky-note-o"
                size={30}
                style={
                  Object {
                    "paddingLeft": 4,
                  }
                }
              />
            </TouchableOpacity>
          </Component>
        </Component>
      </Component>
    </Component>
  </CardView>
</TouchableOpacity>
`;

exports[`Component >>> ItemList Render ItemList 2`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <CardView>
    <Component
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 6,
          "overflow": "hidden",
        }
      }
    >
      <Component
        style={
          Object {
            "flexDirection": "row",
            "paddingBottom": 8,
            "paddingHorizontal": 16,
            "paddingTop": 16,
          }
        }
      >
        <Component
          style={
            Object {
              "width": "80%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "#000",
                "fontSize": 18,
                "fontWeight": "bold",
                "padding": 1,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "#000",
                "fontSize": 16,
              }
            }
          >
             - 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "gray",
                "fontSize": 12,
              }
            }
          >
            sent_on
            1995-12-25 00:00:00
          </Text>
        </Component>
        <Component
          style={
            Object {
              "alignItems": "flex-end",
              "width": "20%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#2796ea",
                "fontSize": 10,
                "fontWeight": "bold",
                "paddingHorizontal": 3,
                "textAlign": "right",
              }
            }
          >
            0
          </Text>
        </Component>
      </Component>
      <Component
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#f5f5f5",
            "flexDirection": "column",
            "paddingHorizontal": 10,
          }
        }
      >
        <Component
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingTop": 5,
              "width": "100%",
            }
          }
        >
          <Component
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Component
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#1F6CBC",
                    "borderRadius": 100,
                    "height": 36,
                    "justifyContent": "center",
                    "width": 36,
                  },
                ]
              }
            >
              <Image
                onLoad={[Function]}
                source={
                  Object {
                    "uri": "data:image/gif;base64,",
                  }
                }
                style={
                  Array [
                    Object {
                      "borderRadius": 100,
                      "height": 36,
                      "resizeMode": "cover",
                      "width": 36,
                    },
                    Object {
                      "display": "none",
                    },
                  ]
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "display": "flex",
                  }
                }
              />
            </Component>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000",
                  "paddingHorizontal": 8,
                  "paddingVertical": 6,
                }
              }
            >
               
            </Text>
          </Component>
          <Component
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <Component
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 3,
                    "flexDirection": "row",
                    "margin": 1,
                    "padding": 3,
                  },
                  Object {
                    "backgroundColor": "#E14333",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  }
                }
              >
                lists_activity_due_date_overdue
              </Text>
            </Component>
          </Component>
        </Component>
        <Component
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "paddingTop": 5,
                "width": "100%",
              },
              Object {
                "paddingTop": 2,
              },
            ]
          }
        >
          <Component
            style={
              Object {
                "alignItems": "center",
                "display": "none",
              }
            }
          >
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingTop": 5,
                }
              }
            >
              <Icon
                allowFontScaling={false}
                name="check-circle"
                size={12}
                style={
                  Object {
                    "color": "#1FBC99",
                    "fontSize": 24,
                  }
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#1FBC99",
                      "justifyContent": "flex-start",
                      "paddingHorizontal": 8,
                      "paddingVertical": 6,
                    },
                    Object {
                      "color": "#1FBC99",
                    },
                  ]
                }
              >
                saved_on_Draft_new_activity_string
              </Text>
            </Component>
          </Component>
          <Component
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Icon
                allowFontScaling={false}
                color="#000"
                name="history"
                size={30}
                style={
                  Object {
                    "paddingLeft": 4,
                  }
                }
              />
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Icon
                allowFontScaling={false}
                color="#000"
                name="sticky-note-o"
                size={30}
                style={
                  Object {
                    "paddingLeft": 4,
                  }
                }
              />
            </TouchableOpacity>
          </Component>
        </Component>
      </Component>
    </Component>
  </CardView>
</TouchableOpacity>
`;

exports[`Render Card 1`] = `
<View
  elevation={3}
  style={
    Object {
      "backgroundColor": "#FFF",
      "borderColor": "#365173",
      "borderRadius": 6,
      "borderWidth": 1,
      "flex": 1,
      "margin": 3,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 2,
    }
  }
/>
`;

exports[`Render FlatButton 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#FFFFFF",
        "borderRadius": 50,
        "borderWidth": 2,
        "paddingHorizontal": 50,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFFFFF",
          "fontWeight": "bold",
        }
      }
    >
      test
    </Text>
  </Component>
</TouchableOpacity>
`;

exports[`Render NavigatorButton 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <Component
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingTop": 5,
      }
    }
  >
    <Component
      style={
        Object {
          "paddingLeft": 15,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        name="sign-out"
        size={12}
        style={
          Object {
            "color": "#FFF",
            "fontSize": 24,
          }
        }
      />
    </Component>
    <Component
      style={
        Object {
          "justifyContent": "flex-start",
          "paddingLeft": 18,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "margin": 5,
          }
        }
      >
        test
      </Text>
    </Component>
  </Component>
</TouchableOpacity>
`;

exports[`Render NavigatorTitle 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "paddingBottom": 5,
      "paddingTop": 5,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "margin": 5,
        }
      }
    >
      test
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "borderColor": "#d6d7da",
            "borderRadius": 4,
            "borderWidth": 1,
            "color": "white",
            "height": 30,
            "padding": 5,
            "textAlign": "center",
            "width": 60,
          },
          Object {
            "borderWidth": 1,
          },
        ]
      }
    >
      5
    </Text>
  </View>
</View>
`;

exports[`Render Swiper 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#1F6CBC",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-around",
        "paddingVertical": 16,
      },
    ]
  }
>
  <View
    style={
      Object {
        "height": 60,
        "justifyContent": "center",
        "margin": 16,
        "width": 240,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../App/Assets/images/logo_white.png",
        }
      }
      style={
        Object {
          "resizeMode": "contain",
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;
